CMakeLists.txt

```diff
@@ -7,9 +7,11 @@ INCLUDE (subdirs)
 INCLUDE (openssl)
 INCLUDE (version)
 
-OPTION (BUILD_BENCHMARK "Build benchmark" OFF)
-OPTION (BUILD_TESTS "Build tests" OFF)
-OPTION (BUILD_SHARED_LIBS "Build shared libs" OFF)
+IF (NOT CHCPP_EMBEDDED_BUILD)
```

Comment:
Also, please consider changing a name to `CLICKHOUSE_CPP_EMBEDDED_BUILD`, like we have in `3.0`: https://github.com/ClickHouse/clickhouse-cpp/blob/3.0/CMakeLists.txt

Reply:
Thank you for answers and pieces of advice. I found another solution. It is not visible to user and handled automatically by build system.
I think, it will be better than new option.